From 6b150b3e9f1d16e6c374e178107ab490a18346dd Mon Sep 17 00:00:00 2001
From: Asad Mehmood <asad78611@googlemail.com>
Date: Thu, 17 Oct 2013 16:05:59 +0100
Subject: [PATCH] monitor: expose min-backlight-step

Expose min-backlight-step so that gnome-settings-daemon can
support backlights with less than 10 steps without mutter
normalizing the brightness back to its original value
---
 src/core/monitor.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/monitor.c b/src/core/monitor.c
index 9ec0c2a..6d65f9e 100644
--- a/src/core/monitor.c
+++ b/src/core/monitor.c
@@ -806,6 +806,9 @@ meta_monitor_manager_handle_get_resources (MetaDBusDisplayConfig *skeleton,
                              g_variant_new_take_string (make_display_name (manager, output)));
       g_variant_builder_add (&properties, "{sv}", "backlight",
                              g_variant_new_int32 (output->backlight));
+      g_variant_builder_add (&properties, "{sv}", "min-backlight-step",
+                             g_variant_new_int32 ((output->backlight_max - output->backlight_min) ?
+                                                  100 / (output->backlight_max - output->backlight_min) : -1));
       g_variant_builder_add (&properties, "{sv}", "primary",
                              g_variant_new_boolean (output->is_primary));
       g_variant_builder_add (&properties, "{sv}", "presentation",
-- 
1.8.4.1

